qiskit/qpy/__init__.py

```diff
@@ -100,6 +100,14 @@
 by ``num_circuits`` in the file header). There is no padding between the
 circuits in the data.
 
+.. _qpy_version_6:
+
+Version 6
+=========
+In Version 6, the symbolic library pulses changed from complex `amp` representation to float (`amp`
+,`angle`) representation. To accommodate this change, when a QPY file of version 5 or lower is loaded,
+these library pulses are converted to the new format.
```

Comment:
```suggestion

Version 6 has an internal update to :ref:`qpy_schedule_symbolic_pulse` loader of `previous` QPY 
version to adapt in the latest Qiskit library pulse representation. In Qiskit Terra 0.23 and above, 
complex `amp` value representation is replaced with float (`amp`, `angle`) pair.
Because a QPY binary file dumped by the QPY version 5 and below implies the data
is still represented by the conventional complex amp format, the loaded pulse parameters
and envelope are immediately converted into new format to instantiate the pulse in new style.
```

@mtreinish Do you have any idea of the better description for this? This is bit tricky update. Conventionally we make some update to the QPY data structure and that requires us to bump the QPY version. However, we don't have any update on the QPY data structure, but pulse class in Terra has been updated with this PR. Since QPY is agnostic to Terra version, we need to bump the QPY version instead.

Reply:
Unless I'm misreading something do we need to bumpy the qpy format version at all. It looks like there isn't a change in the format specification but you're using the qpy format version to indicate how to recreate a pulse from before this PR. Could you use the qiskit version information in the global header instead: https://qiskit.org/documentation/apidoc/qpy.html#qpy-format and then we don't need to bump the format version like this. This kind of situation was part of why I had the qiskit version separate from the format version in the header.

Reply:
Thank you! I'll follow your suggestion.